src/abstractinterpretation.jl

```diff
@@ -247,6 +247,17 @@ function CC.abstract_call_method(interp::JETInterpreter, method::Method, @nospec
         end
     end
 
+    if method.name === :iterate
+        v, _, _ = ret
+        ret_type = (isa(v, Const) ? Core.Typeof(v.val) :
+                    isa(v, Core.PartialStruct) ? v.typ :
+                    isa(v, DataType) || isa(v, Union) ? v :
+                    return)::Type
+        if ret_type !== Union{Nothing, Tuple{Int64, Int64}}
+            report!(interp, InfiniteIterationErrorReport(interp, sv, ret_type))
+        end
+    end
+
```

Comment:
I'm not sure if this is the ideal method to perform this check in but it seems to work